calico_node/startup/startup.go

```diff
@@ -828,6 +833,52 @@ func ensureNodeFelixConfig(c *client.Client, host, key, def string) error {
 	}
 }
 
+// updateClusterType ensures that the ClusterType in the Felix global
+// config has the data contained in the passed in clusterType.  This includes
+// merging what was already set in ClusterType with the passed in clusterType.
+func updateClusterType(c *client.Client, key, clusterType string) error {
+	// If no data is passed in to add, then nothing to do.
+	if clusterType == "" {
+		return nil
+	}
+	global := ""
+	if val, assigned, err := c.Config().GetFelixConfig("ClusterType", global); err != nil {
+		return err
+	} else if !assigned {
+		return c.Config().SetFelixConfig(key, global, clusterType)
+	} else if val == "" {
+		return c.Config().SetFelixConfig(key, global, clusterType)
+	} else {
+		// If were here then bot val and clusterType have data, this means
+		// the Split will not have any empty string values
+
+		updateNeeded := false // keep track if we add any elements
+		cts := strings.Split(val, ",")
+		for _, ct := range strings.Split(clusterType, ",") {
+			found := false
+			for _, x := range cts {
+				if ct == x {
+					found = true
```

Comment:
I was on the fence originally about using the `continue NEXT` so I didn't mind getting rid of it.  It feels like 'goto's in c/c++ which was always something to avoid if possible so I think that sticks with me.  Also this is pretty simple so I don't think it really helps with the readability much.  I'm going to leave it as is.